daemon/libnetwork/sandbox_dns_unix_test.go

```diff
@@ -90,3 +95,69 @@ func TestDNSOptions(t *testing.T) {
 	dnsOptionsList = getResolvConfOptions(t, sb2.config.resolvConfPath)
 	assert.Check(t, is.DeepEqual([]string{"ndots:0"}, dnsOptionsList))
 }
+
+func TestNonHostNetDNSRestart(t *testing.T) {
+	c, err := New(context.Background(), config.OptionDataDir(t.TempDir()))
+	assert.NilError(t, err)
+
+	// Step 1: Create initial sandbox (simulating first container start)
+	sb, err := c.NewSandbox(context.Background(), "cnt1")
+	assert.NilError(t, err)
+
+	sb.startResolver(false)
+
+	err = sb.setupDNS()
+	assert.NilError(t, err)
+	err = sb.rebuildDNS()
+	assert.NilError(t, err)
+
+	// Step 2: Simulate cri-dockerd modifying the resolv.conf for a Kubernetes pause container.
+	// This mimics the behavior where external tools (like cri-dockerd) customize DNS
+	// settings for K8s pods, which should be preserved during container restart/unpause.
```

Comment:
I'd prefer not to mention that very specific use-case. To be honest, it still makes little sense to me that you're trying to work around some of the k8s primitives in such a way, but I'm not going to fight over that.

In general, I think this enhancement makes sense though and we got another  request for that a few days ago. See https://github.com/moby/moby/issues/51570.

Could you please rephrase that in more general terms? Something like 'Step 2: Simulate user manually overwritting the container's resolv.conf'?

This would avoid future maintainers raising an eyebrow at this comment wondering what's the real use case.